barretenberg/cpp/src/barretenberg/polynomials/backing_memory.hpp

```diff
@@ -126,12 +145,59 @@ template <typename T> class FileBackedMemory : public BackingMemory<T> {
 
         // Set file size
         if (ftruncate(fd, static_cast<off_t>(file_size)) != 0) {
+            close(fd);
+            std::filesystem::remove(filename);
             throw_or_abort("Failed to set file size");
         }
```

Comment:
this looks like its significantly more hacky than just a two flag approach replacing the old runtime slow low mem prover:
- RAM budget
- disk budget
by default we have unlimited RAM budget (represented as 0) and 0 disk budget. we use disk first (i realized we cant use it after memory budget, because we will temporarily need more RAM), then we use memory. we just track with a global variable. I dont trust OS's to be consistent in telling us how much usage otherwise

Reply:
I'm happy to try the budget approach too.